### ISSUE TITLE
Cannot work on kernelsu's device

### ISSUE THREAD

**Issue by kissunyeason** [skittles9823/QuickSwitch#63]
提示需要 magisk，![Screenshot_20240531-112106_QuickSwitch.png](https://github.com/skittles9823/QuickSwitch/assets/17369666/205b74f6-7ac0-452e-9549-5d8e00444d2d)

![Screenshot_20240531-112049_KernelSU.png](https://github.com/skittles9823/QuickSwitch/assets/17369666/72676206-f28b-483b-993b-99f875cbd2d4)

**Comment by MoonLeeeaf** [comment]
```sh
# 首先你得安装相应的模块
export KSU=true # 让 QuickSwitch 知道你在用 Kernel Su
/data/adb/modules/quickswitch/quickswitch --ch=app.lawnchair # 最近任务提供应用的包名，根据实际填写
reboot
# 然后看看最近任务是否有所改变
```

**Comment by Obnomus** [comment]
@MoonLeeeaf I tried this solution but still I got the error.

**Comment by MoonLeeeaf** [comment]
@Obnomus I'm sorry that I've used another way to change the default RecentsProvider and no longer used QuickSwitch. But if you use Lawnchair too, maybe a module made by myself will help you😀

**Comment by MoonLeeeaf** [comment]
Try using it?
https://github.com/MoonLeeeaf/Recents_Provider_changer_for_Lawnchair/releases/download/v1.0.0/QuickSwitch_Lawnchair.zip

**Comment by Obnomus** [comment]
> Try using it? https://github.com/MoonLeeeaf/Recents_Provider_changer_for_Lawnchair/releases/download/v1.0.0/QuickSwitch_Lawnchair.zip

So i can use this module instead of quickswitch

**Comment by Obnomus** [comment]
I already got the pixel launcher as recent provider but I can't get the animations.

**Comment by MoonLeeeaf** [comment]
> > Try using it? https://github.com/MoonLeeeaf/Recents_Provider_changer_for_Lawnchair/releases/download/v1.0.0/QuickSwitch_Lawnchair.zip
> 
> So i can use this module instead of quickswitch

@Obnomus No, it's impossible. But you can try replacing Lawnchair.apk to another launcher you like, then replacing the package name🤔

**Comment by MoonLeeeaf** [comment]
> I already got the pixel launcher as recent provider but I can't get the animations.

I'm sorry that I can't give any suggestions to solve it :(
What's the OS of your phone?

**Comment by Obnomus** [comment]
> > I already got the pixel launcher as recent provider but I can't get the animations.
> 
> I'm sorry that I can't give any suggestions to solve it :( What's the OS of your phone?

I'm on custom rom first is android 14 qpr3 and other one is on android 15

**Comment by MoonLeeeaf** [comment]
> > > I already got the pixel launcher as recent provider but I can't get the animations.
> > 
> > 
> > I'm sorry that I can't give any suggestions to solve it :( What's the OS of your phone?
> 
> I'm on custom rom first is android 14 qpr3 and other one is on android 15

🤔🤔
除了没有动画 其他正常吗
Except for no animation, is everything else normal?
模块基于 Overlay 修改默认的 Recents Provider, 但可能会随着安卓版本的差异而有不同情况
The module modifies the default Recents Provider based on Overlay, but it may vary depending on the Android version.
很抱歉 但是我确实没有什么想法 试试找别的大佬吧 :)
Sorry, but I really don't have any ideas about it. Try asking other heavyweights for help :)

**Comment by Obnomus** [comment]
> > > > I already got the pixel launcher as recent provider but I can't get the animations.
> > > 
> > > 
> > > I'm sorry that I can't give any suggestions to solve it :( What's the OS of your phone?
> > 
> > 
> > I'm on custom rom first is android 14 qpr3 and other one is on android 15
> 
> 🤔🤔 除了没有动画 其他正常吗 Except for no animation, is everything else normal? 模块基于 Overlay 修改默认的 Recents Provider, 但可能会随着安卓版本的差异而有不同情况 The module modifies the default Recents Provider based on Overlay, but it may vary depending on the Android version. 很抱歉 但是我确实没有什么想法 试试找别的大佬吧 :) Sorry, but I really don't have any ideas about it. Try asking other heavyweights for help :)

Everything works as it should be also recent provider changed to what I wanted, but quickswitch app shows "QuickSwitch requires Magisk root access to work properly" that's the only problem and I just want to know why it's happening

**Comment by MoonLeeeaf** [comment]
> > ...
> Everything works as it should be also recent provider changed to what I wanted, but quickswitch app shows "QuickSwitch requires Magisk root access to work properly" that's the only problem and I just want to know why it's happening

Oh, maybe the "conflicting modules check" caused it
It's hard for us to understand the code :(

Lxyz/paphonb/quickstepswitcher/MainActivity;->b0(Lxyz/paphonb/quickstepswitcher/MainActivity;)V
![Screenshot_20241020-161532_MT管理器](https://github.com/user-attachments/assets/f7ab06bd-9973-4921-9372-413d501e38ba)

Lxyz/paphonb/quickstepswitcher/MainActivity;->d0(Z)V
Resource ID: 7f110059 (...Magisk root to...)
![Screenshot_20241020-161658_MT管理器](https://github.com/user-attachments/assets/9c9a8d06-1aea-4694-9a0c-bb5f3726e2d8)

I'm not the developer of this project, this explaining may be wrong🤔

"su" command maybe also cause this😱
![Screenshot_20241020-162234_MT管理器](https://github.com/user-attachments/assets/c20be5dd-e990-4c75-b39f-95234f2fe462)

All in all, this Magisk module is still not stable. The developers can't think about everything maybe happen. But it gives us a good idea to do like QuickSwitch :)

tl;dr: 能正常用的话就可以不用管别的了,开发人员没有考虑到其他复杂的原因,这个模块仍需要被完善
If it works normally, you don't have to worry about anything else. The developer has not considered other complicated reasons, and this module still needs to be improved.
